stackdriver.go

```diff
@@ -255,6 +255,23 @@ type Options struct {
 	// to Stackdriver Monitoring. This is only used for Proto metrics export
 	// for now. The minimum number of workers is 1.
 	NumberOfWorkers int
+
+	// ResourceByDescriptor may be provided to supply monitored resource dynamically
+	// based on the metric Descriptor. Most users will not need to set this,
+	// but should instead set ResourceDetector.
+	//
+	// The MonitoredResource and ResourceDetector fields are ignored if this
+	// field is set to a non-nil value.
+	//
+	// The ResourceByDescriptor is called to derive monitored resources from
+	// metric.Descriptor and label map associated with the metric. If any label is
+	// used for the derived resource then it will be removed the label map.
+	// Returned label map would then be a subset of the original map.
```

Comment:
Could you describe what the return label maps are used for?

Reply:
added the description.